src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -302,14 +302,25 @@ def disable_raycluster_tls(resources):
             ]
         )
     ):
-        if i == 1:
+        if (
+            generic_template_spec["workerGroupSpecs"][0]["template"]["spec"][
+                "initContainers"
+            ][i]["name"]
+            == "create-cert"
+        ):
             del generic_template_spec["workerGroupSpecs"][0]["template"]["spec"][
                 "initContainers"
             ][i]
 
-    if len(resources["GenericItems"]) >= 3:
-        del resources["GenericItems"][3]  # rayclient route
-        del resources["GenericItems"][2]  # ca-secret
+    updated_items = []
+    for i in resources["GenericItems"][:]:
+        if "rayclient-deployment-name" in i["generictemplate"]["metadata"]["name"]:
+            continue
+        if "ca-secret-deployment-name" in i["generictemplate"]["metadata"]["name"]:
+            continue
+        updated_items.append(i)
+
+    resources["GenericItems"] = updated_items
```

Comment:
Why aren't you deleting these items? @Bobbins228

Reply:
It's essentially the same right? 
`resources["GenericItems"]` is equated to `resources["GenericItems"]` without the ca-secret and ray-client. 

I tested this out and compared the yaml file with one generated with the current sdk and there were no differences. 

I could do what I did [here](https://github.com/project-codeflare/codeflare-sdk/pull/244/files#diff-279fd895cfcb8295f1b610826f8eb7a40fd543ff873661acd9dc452cef3d9d14R305-R313) if necessary.